### ISSUE TITLE
Memory error when creating labels for layer with big values in spacing. 

### ISSUE THREAD

**Issue by Czaki** [napari/napari#1407]
## 🐛 Bug
When image layers have big spacing then adding labels layer with button ends with error because of try allocation of TiB memory

This is relay important because works to introduce metadata parsing and extraction of spacing information are in #1384 then this problem may affect more people.

## To Reproduce

1. Load 3d image. 
2. Set layer spacing to something big (like 300, 70 ,70)
3. Click add button layer. 

```
Traceback (most recent call last):
  File "/Users/gbokota/.pyenv/versions/partseg3.6/lib/python3.6/site-packages/napari/_qt/qt_viewer_buttons.py", line 49, in <lambda>
    lambda: self.viewer._new_labels(),
  File "/Users/gbokota/.pyenv/versions/partseg3.6/lib/python3.6/site-packages/napari/components/viewer_model.py", line 308, in _new_labels
    empty_labels = np.zeros(dims, dtype=int)
MemoryError: Unable to allocate 357. TiB for an array with shape (1, 8700, 65536, 86016) and data type int64
```
## Expected behavior
Labels layer should have similar voxel size like image layer.  

## Environment

 - Please copy and paste the information at napari info option in help menubar here:
napari: 0.3.5
Platform: Linux-5.4.0-40-generic-x86_64-with-debian-bullseye-sid
Python: 3.6.8 (default, Jun 26 2020, 19:49:46) [GCC 9.3.0]
Qt: 5.14.2
PyQt5: 5.14.2
NumPy: 1.19.0
SciPy: 1.5.0
Dask: 2.19.0
VisPy: 0.6.4

GL version: 4.6.0 NVIDIA 390.138
MAX_TEXTURE_SIZE: 32768

Plugins:
- PartSeg: 0.11.0b7
- napari-plugin-engine: 0.1.6
- napari-plugins: 0.1.0
- svg: 0.1.3
 - Any other relevant information:

PartSeg plugin is only reader plugin 
## Additional context

I try to think about solution and think that there could be asllo problem if voxel size is something like (0.3, 0,07, 0,07) because label voxel will be bigger than image voxel. 

Possible solution is to divede size of layer by minimum voxel size (spacing) of existing layers. Other option is to take voxel size from active layer.  But maybe you will have better idea So I open Issue instead of PR.

**Comment by sofroniewn** [comment]
Yeah, good catch, sorry about that. I should have a fix for that in #1360 or #1276. It relates to us not having a proper world coordinates system!!

**Comment by Czaki** [comment]
Thanks. It will be fixed in next release?

**Comment by sofroniewn** [comment]
I'll try. If not next release than fairly confident it will be the one after.

**Comment by sofroniewn** [comment]
Closed by #1360
